### ISSUE TITLE
Document with nested class executing parent class method

### ISSUE THREAD

**Issue by kerwynrg** [Automattic/mongoose#5227]
I don't know if I'm doing something wrong, I couldn't find any related solution.
Mongoose: 4.9.7
Node: 7.9.0
MongoDB: 3.4.3

I have 2 classes
``` javascript
import { Model } from 'mongoose';
export default class BaseModel extends Model {
  getString(): string {
    return `parent: ${this.id}`;
  }
}
```
// Different file
```javascript
class User extends BaseModel {
  getFullname (): string {
    return `${this.name} ${this.lastname}`;
  }

  getString (): string {
    return `${this.username} - ${this.getFullname()}`;
  }
}
```

And I have this document in MongoDB
```
{
    "_id" : ObjectId("1"),
    "username" : "username",
    "name" : "Name",
    "lastname" : "Lastname"
}
```

```javascript
export default class BaseRepository {
  modelClass: typeof BaseModel;
  constructor (modelClass: typeof BaseModel) {
    // At this point modelClass is the User class extending BaseModel
    this.modelClass = modelClass;
  }

  getOneById (id: string): Object {
    return this.modelClass.findById(id).exec()
    .then((result) => {
      return result;
    });
  }
}
```

When executing method getOneById is returning in GraphiQL
```json
{
  "_id": "1",
  "username" : "username",
  "name" : "Name",
  "lastname" : "Lastname",
  "getFullname": "Kerwyn Jair Rojas Garcia",
  "getString": "parent: 590cc0e400dd92a3f4f6206b"
}
```

Why?
It supposed to return the value of getString method from User inherited class
```javascript
getString (): string {
    return `${this.username} - ${this.getFullname()}`;
  }
```
Instead is returning the result from parent method.

If I remove the parent method (In BaseModel class)
It works correctly.

For me is important to have a parent class so I can implement some features like caching results or documents.

This Stackoverflow  question looks like the same problem.
http://stackoverflow.com/questions/43343378/mongoose-model-as-an-es6-class-with-inheritance-chain

Thanks in advance.

**Comment by vkarpov15** [comment]
How are you creating your mongoose models? I don't see that anywhere in your example

**Comment by kerwynrg** [comment]
This is the full user file
```javascript
/* @flow */
import {
  BaseModel,
  modelCreator
} from 'utils';
import schema from './user.schema';

const collectionName = 'users';

class User extends BaseModel {
  getFullname (): string {
    return `${this.name} ${this.lastname}`;
  }

  toString (): string {
    return `${this.username} - ${this.getFullname()}`;
  }
}

export default modelCreator(User, schema, collectionName);
```

This is the modelCreator file
```javascript
import mongoose, {
  Schema
} from 'mongoose';

const declareEvents = (schema: Schema) => {
  // Some logic
};

export default (modelClass: Function, schema: Schema, collectionName: string): Function => {
  declareEvents(schema);
  return mongoose.model(modelClass, schema, collectionName);
};
```

Thanks in advance

**Comment by vkarpov15** [comment]
Thanks for reporting, fix will be in 4.10.3

**Comment by kerwynrg** [comment]
Thanks 👍
